tests/task/delay_file_delete_test.php

```diff
@@ -0,0 +1,78 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+
+namespace tool_objectfs\task;
+
+/**
+ * Task that checks for files that were delayed for deletion.
+ * It deletes the files which have an expired delay time.
+ *
+ * @package   tool_objectfs
+ * @author    Kateryna Martynenko <kateryna.martynenko@moodle.com>
+ * @copyright Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once(__DIR__ . '/../../lib.php');
+
+class delay_file_delete_test extends \tool_objectfs\tests\testcase  {
+
+    /** @var string $stringname */
+    protected $stringname = 'delay_delete_external_objects_task';
+
+    /**
+     * Execute task
+     */
+    public function execute() {
```

Comment:
I don't understand this test. How is it validating that elements are being deleted from the external system? I think we should [create mocks](https://subscription.packtpub.com/book/application-development/9781782169581/1/ch01lvl1sec18/using-mock-objects-(simple)) for the method `$filesystem->delete_external_file_from_hash(...)`, and check that when the task is run, the method is called under the correct circumstances.